src/components/apps/RatingWidget.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @author sriram
+ *
+ * A widget that allows users to rate an app.
+ */
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "i18n";
+import { queryCache, useMutation, useQuery } from "react-query";
+
+import { Rate } from "@cyverse-de/ui-lib";
+
+import constants from "../../constants";
+import { getAppById, rateApp, APP_BY_ID_QUERY_KEY } from "serviceFacades/apps";
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+
+import { Grid, Typography } from "@material-ui/core";
+
+export default function RatingWidget(props) {
+    const { appId, systemId, appName } = props;
+    const { t } = useTranslation("apps");
+
+    const [ratingMutationError, setRatingMutationError] = useState(null);
+    const [selectedApp, setSelectedApp] = useState(null);
+    const [appRating, setAppRating] = useState({
+        average: 0,
+        total: 0,
+        user: 0,
+    });
+
+    const { isFetching: appByIdStatus, error: appByIdError } = useQuery({
+        queryKey: [APP_BY_ID_QUERY_KEY, { systemId, appId }],
+        queryFn: getAppById,
+        config: {
+            enabled: appId != null && systemId !== null,
+            onSuccess: (result) => {
+                setSelectedApp(result?.apps[0]);
+            },
+        },
+    });
+
+    const [rating, { status: ratingMutationStatus }] = useMutation(rateApp, {
+        onSuccess: () => {
+            queryCache.invalidateQueries([
+                APP_BY_ID_QUERY_KEY,
+                { systemId, appId },
+            ]);
+        },
+        onError: (e) => {
+            setRatingMutationError(e);
+        },
+    });
+
+    const onRatingChange = (event, value) => {
+        rating({
+            appId,
+            systemId,
+            rating: value,
+        });
+    };
+
+    const onDeleteRating = () => {
+        rating({
+            appId,
+            systemId,
+            rating: null,
+        });
+    };
+
+    useEffect(() => {
+        if (selectedApp) {
+            setAppRating(selectedApp.rating);
+        }
+    }, [selectedApp]);
+
+    const loading = appByIdStatus || ratingMutationStatus === constants.LOADING;
```

Comment:
I think this was meant to check both variables for the loading status:
```suggestion
    const loading = appByIdStatus === constants.LOADING || ratingMutationStatus === constants.LOADING;
```

This should also work:
```
import isQueryLoading from "components/utils/isQueryLoading";

const loading = isQueryLoading([appByIdStatus, ratingMutationStatus])
```

Reply:
The `appByIdStatus` var is actually an alias for the app listing query's `isFetching` flag, so this works, though maybe the alias is misnamed. The `isQueryLoading` util function is still a good idea 👍